src/ATen/native/xpu/SummaryOps.cpp

```diff
@@ -21,4 +22,30 @@ Tensor XPUNativeFunctions::bincount(
   return native::xpu::bincount_kernel(self, weights, minlength);
 }
 
+Tensor XPUNativeFunctions::histc(
+    const Tensor& self,
+    int64_t nbins,
+    const Scalar& min,
+    const Scalar& max) {
+  if (self.scalar_type() == ScalarType::Half) {
+    AT_ERROR("HalfTensor is not supported");
+  }
+  // See Note [Writing Nondeterministic Operations]
```

Comment:
let's move deterministic check to kernel level, which is related to kernel algorithm